zips/zip-0312.rst

```diff
@@ -126,60 +131,165 @@ Specification
+  representation and Scalar representation, respectively.
+* Let functions $\mathsf{Scalar} \;{\small ⦂}\; \{ 0 .. p-1 \} \rightarrow \mathtt{Scalar}$,
+  $\mathsf{RandomScalar} \;{\small ⦂}\; () \stackrel{\tiny\mathrm{R}}{\rightarrow} \mathtt{Scalar}$,
+  $\mathtt{ScalarMult} \;{\small ⦂}\; \mathtt{Element} \times \mathtt{Scalar} \rightarrow \mathtt{Element}$,
+  $\mathsf{ScalarBaseMult} \;{\small ⦂}\; \mathtt{Scalar} \rightarrow \mathtt{Element}$,
+  $\mathsf{SerializeElement} \;{\small ⦂}\; \mathtt{Element} \rightarrow \mathbb{B}^{{\kern-0.05em\tiny\mathbb{Y}}[\mathtt{Ne}]}$,
+  $\mathsf{DeserializeElement} \;{\small ⦂}\; \mathbb{B}^{{\kern-0.05em\tiny\mathbb{Y}}[\mathtt{Ne}]} \rightarrow \mathtt{Element} \cup \{ \bot \}$,
+  $\mathsf{SerializeScalar} \;{\small ⦂}\; \mathtt{Scalar} \rightarrow \mathbb{B}^{{\kern-0.05em\tiny\mathbb{Y}}[\mathtt{Ns}]}$,
+  $\mathsf{DeserializeScalar} \;{\small ⦂}\; \mathbb{B}^{{\kern-0.05em\tiny\mathbb{Y}}[\mathtt{Ns}]} \rightarrow \mathtt{Scalar} \cup \{ \bot \}$
+  also be as described in [#frost-primeordergroup]_ (where "raising an error"
+  corresponds to returning $\bot$).
+
+Note that the notation for elliptic-curve scalar multiplication,
+``G.ScalarMult(P, k)``, differs from that used in the Zcash Protocol
+Specification, which would use $[k]\, P$ with the group implied by $P$
+[#protocol-abstractgroup]_.
 
 
 Key Generation
 --------------
 
-While key generation is out of scope for this ZIP and the FROST spec [#FROST]_,
-it needs to be consistent with FROST, see [#frost-tdkg]_ for guidance. The spend
-authorization private key $\mathsf{ask}$ [#protocol-spendauthsig]_ is the
-particular key that must be used in the context of this ZIP. Note that the
-$\mathsf{ask}$ is usually derived from the spending key
-$\mathsf{sk}$, though that is not required. Not doing so allows using
-distributed key generation, since the key it generates is unpredictable. Note
-however that not deriving $\mathsf{ask}$ from $\mathsf{sk}$ prevents
-using seed phrases to recover the original secret (which may be something
-desirable in the context of FROST).
+FROST key generation with a trusted dealer is described in [#frost-tdkg]_. That
+description does not support key re-randomization but can be adapted to do so,
+as described in section 5 of [#frost-rerandomized]_.
+
+However, trusted dealer key generation has the severe disadvantage that it
+requires absolutely trustworthy hardware and software to be used by the dealer
+in order to reliably avoid exposing the private key material. That is, it is not
+sufficient simply to have an individual or organization that is trusted by all
+participants; their entire hardware and software stack also needs to be trusted
+by all participants. Since this will often be an unrealistic assumption that
+partly defeats the point of using a threshold signature scheme, trusted dealer
+key generation SHOULD NOT be used.
```

Comment:
```suggestion
Trusted dealer key generation has the severe disadvantage that it
requires absolutely trustworthy hardware and software to be used by the dealer
in order to reliably avoid exposing the private key material. That is, it is not
sufficient simply to have an individual or organization that is trusted by all
participants; their entire hardware and software stack also needs to be trusted
by all participants. Since this will often be an unrealistic assumption that
partly defeats the point of using a threshold signature scheme, trusted dealer
key generation SHOULD NOT be used.
```

And make this a WARNING section